### PR TITLE
Upgrade typescript to 2.1

### PR DESCRIPTION
**Pull request by maxlang**
cc @tamird

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/12293)

### CODE REVIEW

File: pkg/ui/yarn.lock [hunk 1]
```diff
@@ -3312,7 +3312,11 @@ typedarray@~0.0.5:
   version "0.0.6"
   resolved "https://registry.yarnpkg.com/typedarray/-/typedarray-0.0.6.tgz#867ac74e3864187b1d3d47d996a78ec5c8830777"
 
-typescript@^2.0.3, typescript@^2.0.9:
+typescript:
```
Comment (mrtracy) [inline pkg/ui/yarn.lock]: There were previously two locks, for typescript ^2.0.3 and ^2.0.9.   The locks don't match up, i'm not quite sure which tools are using different versions of typescript; although apparently, they were already using 2.1.1.

**Comment by mrtracy** [comment]
This needs to be installed into JSPM as well, or it won't affect our compiled code.

**Comment by tamird** [comment]
As @mrtracy points out, this is probably not having any effect.

**Comment by maxlang** [comment]
Oops this was just an issue I was having with visual studio. Pls ignore.

**Comment by tamird** [comment]
Don't we want to upgrade, though?

**Comment by maxlang** [comment]
Yeah, I looked into this a bit more but I'm running into an issue with the way we import CSS. Typescript 2.1.4 complains on lines like

``` js
import "nvd3/build/nv.d3.min.css!";
import "react-select/dist/react-select.css!";
import "build/app.css!";
```

which https://github.com/systemjs/plugin-css expects. I'm going to try this again later.

**Comment by tamird** [comment]
What's the complaint?

**Comment by maxlang** [comment]
```
jspm bundle-sfx app/app build/app.js
     Building the single-file sfx bundle for app/app...

err  Error on translate for typings/modules/sinon/index.d.ts!github:frankwallis/plugin-typescript@5.3.1/plugin.js at file:///Users/maxlang/go/src/github.com/cockroachdb/cockroach/pkg/ui/typings/modules/sinon/index.d.ts!file:///Users/maxlang/go/src/github.com/cockroachdb/cockroach/pkg/ui/jspm_packages/github/frankwallis/plugin-typescript@5.3.1/plugin.js
	Loading typings/index.d.ts!github:frankwallis/plugin-typescript@5.3.1/plugin.js
	Loading app/app.tsx
	Error: Debug Failure. False expression: File file:///Users/maxlang/go/src/github.com/cockroachdb/cockroach/pkg/ui/jspm_packages/npm/nvd3@1.8.4/build/nv.d3.min.css has unknown extension.
         at Object.assert (file:///Users/maxlang/go/src/github.com/cockroachdb/cockroach/pkg/ui/jspm_packages/npm/typescript@2.1.4/lib/typescript.js:2750:17)
         at Object.fail (file:///Users/maxlang/go/src/github.com/cockroachdb/cockroach/pkg/ui/jspm_packages/npm/typescript@2.1.4/lib/typescript.js:2755:15)
         at Object.extensionFromPath (file:///Users/maxlang/go/src/github.com/cockroachdb/cockroach/pkg/ui/jspm_packages/npm/typescript@2.1.4/lib/typescript.js:2873:13)
         at file:///Users/maxlang/go/src/github.com/cockroachdb/cockroach/pkg/ui/jspm_packages/npm/typescript@2.1.4/lib/typescript.js:52874:42
         at Array.map (native)
         at resolveModuleNamesWorker (file:///Users/maxlang/go/src/github.com/cockroachdb/cockroach/pkg/ui/jspm_packages/npm/typescript@2.1.4/lib/typescript.js:52869:71)
         at resolveModuleNamesReusingOldState (file:///Users/maxlang/go/src/github.com/cockroachdb/cockroach/pkg/ui/jspm_packages/npm/typescript@2.1.4/lib/typescript.js:53006:18)
         at processImportedModules (file:///Users/maxlang/go/src/github.com/cockroachdb/cockroach/pkg/ui/jspm_packages/npm/typescript@2.1.4/lib/typescript.js:53746:29)
         at findSourceFile (file:///Users/maxlang/go/src/github.com/cockroachdb/cockroach/pkg/ui/jspm_packages/npm/typescript@2.1.4/lib/typescript.js:53667:11)
         at processSourceFile (file:///Users/maxlang/go/src/github.com/cockroachdb/cockroach/pkg/ui/jspm_packages/npm/typescript@2.1.4/lib/typescript.js:53584:23)
         at processRootFile (file:///Users/maxlang/go/src/github.com/cockroachdb/cockroach/pkg/ui/jspm_packages/npm/typescript@2.1.4/lib/typescript.js:53489:9)
         at file:///Users/maxlang/go/src/github.com/cockroachdb/cockroach/pkg/ui/jspm_packages/npm/typescript@2.1.4/lib/typescript.js:52905:18
         at Object.forEach (file:///Users/maxlang/go/src/github.com/cockroachdb/cockroach/pkg/ui/jspm_packages/npm/typescript@2.1.4/lib/typescript.js:1113:24)
         at Module.createProgram (file:///Users/maxlang/go/src/github.com/cockroachdb/cockroach/pkg/ui/jspm_packages/npm/typescript@2.1.4/lib/typescript.js:52904:12)
         at TypeChecker.getAllDiagnostics (file:///Users/maxlang/go/src/github.com/cockroachdb/cockroach/pkg/ui/jspm_packages/github/frankwallis/plugin-typescript@5.3.1/plugin.js:878:48)
         at TypeChecker.check (file:///Users/maxlang/go/src/github.com/cockroachdb/cockroach/pkg/ui/jspm_packages/github/frankwallis/plugin-typescript@5.3.1/plugin.js:824:37)
make: *** [build/app.js] Error 1
```

**Comment by maxlang** [comment]
I pushed more changes to my branch and tried to reopen but nothing seems to have taken...

**Comment by maxlang** [comment]
Remade PR.

**Pull request closed without merging**